apps/web/utils/string.tsx

```diff
@@ -82,27 +98,40 @@ export const parseStringToComponents = (
               </PDFDownloadLinkComponent>
             );
           }
-          case definedTermModal: {
-            // TODO: Matt or Nicholas, add correct section for customHandler
-            const randomSection = Math.floor(Math.random() * 10) + 1;
+
+          case definedTermModal:
             return (
               <Tooltip
-                tooltipContent={GLOSSARY_TERMS.default.tooltipContent}
+                tooltipContent={TooltipGlossaryData.get(tooltipTerm)}
                 triggerContent={
                   <Button
                     variant="glossary"
                     onPress={
-                      customHandler
-                        ? () => customHandler(`9.9.9.${randomSection}`)
-                        : () => {}
+                      customHandler ? () => customHandler(term) : () => {}
                     }
                   >
                     {domToReact(domNode.children as DOMNode[])}
                   </Button>
                 }
               ></Tooltip>
             );
-          }
+
+          case glossaryTerm:
+            return (
+              <span className="ui-ModalSide-Term">
+                {domToReact(domNode.children as DOMNode[]) as string}
+              </span>
+            );
+
+          case buildingCode:
+            return (
+              <ModalSide
+                type={ModalSideDataEnum.BUILDING_CODE}
+                triggerContent={<Button variant="code">{term}</Button>}
+                modalData={BuildingCodeJSONData}
+                scrollTo={"9.9.9.1"}
```

Comment:
I see 9.9.9.1 is hardcoded - Will this implementation work for additional workflows?

Reply:
This will not be hardcoded once the Building code stuff is done. It would "scroll to" the correct building code section